pulsar-client-cpp/include/pulsar/ProducerConfiguration.h

```diff
@@ -185,29 +264,120 @@ class PULSAR_PUBLIC ProducerConfiguration {
     bool getBlockIfQueueFull() const;
 
     // Zero queue size feature will not be supported on consumer end if batching is enabled
+
+    /**
+     * Control whether automatic batching of messages is enabled or not for the producer. <i>Default value:
+     * false (no batching).</i>
+     *
+     * When batching is enabled, multiple calls to Producer::sendAsync can result in a single batch to be sent
+     * to the broker, leading to better throughput, especially when publishing small messages. If compression
+     * is enabled, messages are compressed at the batch level, leading to a much better compression ratio for
+     * similar headers or contents.
+     *
+     * When the default batch delay is set to 10 ms and the default batch size is 1000 messages.
+     *
+     * @see ProducerConfiguration::setBatchingMaxPublishDelayMs
+     *
+     */
     ProducerConfiguration& setBatchingEnabled(const bool& batchingEnabled);
+
+    /**
+     * Return the flag whether automatic message batching is enabled or not for the producer.
+     *
+     * @return true if automatic message batching is enabled. Otherwise it returns false.
+     * @since 2.0.0 <br>
+     *        It is enabled by default.
+     */
     const bool& getBatchingEnabled() const;
 
+    /**
+     * Set the max number of messages permitted in a batch. <i>Default value: 1000.</i> If you set this option
+     * to a value greater than 1, messages are queued until this threshold is reached or batch interval has
+     * elapsed.
+     *
+     * All messages in a batch are published as
+     *      a single batch message. The consumer is delivered individual messages in the batch in the same
```

Comment:
The consumer is delivered with ... or The consumer delivers...

same for the next